src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Symbolic/SymbolicRegexThresholds.cs

```diff
@@ -8,33 +8,36 @@ namespace System.Text.RegularExpressions.Symbolic
     /// </summary>
     internal static class SymbolicRegexThresholds
     {
-        /// <summary>Maximum number of built states before switching over to NFA mode.</summary>
+        /// <summary>Maximum number of <see cref="SymbolicRegexNode{TSet}"/> instances before switching over to NFA mode.</summary>
         /// <remarks>
         /// By default, all matching starts out using DFAs, where every state transitions to one and only one
         /// state for any minterm (each character maps to one minterm).  Some regular expressions, however, can result
         /// in really, really large DFA state graphs, much too big to actually store.  Instead of failing when we
         /// encounter such state graphs, at some point we instead switch from processing as a DFA to processing as
-        /// an NFA.  As an NFA, we instead track all of the states we're in at any given point, and transitioning
-        /// from one "state" to the next really means for every constituent state that composes our current "state",
-        /// we find all possible states that transitioning out of each of them could result in, and the union of
-        /// all of those is our new "state".  This constant represents the size of the graph after which we start
-        /// processing as an NFA instead of as a DFA.  This processing doesn't change immediately, however. All
-        /// processing starts out in DFA mode, even if we've previously triggered NFA mode for the same regex.
-        /// We switch over into NFA mode the first time a given traversal (match operation) results in us needing
-        /// to create a new node and the graph is already or newly beyond this threshold.
+        /// an NFA. As an NFA, we instead track all of the states we're in at any given point.
         /// </remarks>
-        internal const int NfaThreshold = 10_000;
+        /// <remarks>
+        /// This limit is chosen due to memory usage constraints, the largest possible memory allocation for a regex instance
+        /// is currently approx. 50 MB.
```

Comment:
just curious, does 50MB = N x {some size of struct/object/array} ? if so might be worth noting?

Reply:
oh never mind it's below.

Reply:
right, so the 50MB divided by 400bytes roughly per symbolic regex node gives the total node boundary 
125k that is also a bit of an over approximation because a single symbolic regex node occupies less than 400 bytes but states contain nodes + a few more fields so using 400 is pretty good over approximation to count for that difference.